File: greedy/1481_least_number_of_unique_integers_after_K_removals.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findLeastNumOfUniqueInts = function(arr, k) {

    let freq = {};

    for(let a of arr){
        freq[a] = (freq[a] || 0) + 1;
    }

    let vals = [];

    for(let k of Object.keys(freq)){
        vals.push(freq[k]);
    }

    vals.sort((a,b) => a - b);

    let remove = 0;
    for(let v of vals){
        if(v <= k){
            k -= v;
            remove ++
        } else {
            break
        }
    }

    return vals.length - remove

};

File: greedy/1481_least_number_of_unique_integers_after_K_removals.rs
use std::collections::HashMap;

impl Solution {
    pub fn find_least_num_of_unique_ints(arr: Vec<i32>, k: i32) -> i32 {

        let mut freq:HashMap<i32, i32> = HashMap::new();

        for a in arr {
            freq
                .entry(a)
                .and_modify(|c| *c += 1)
                .or_insert(1);
        }

        let mut counts: Vec<i32> = Vec::new();

        for (i,v) in freq.iter(){
            counts.push(*v);
        }

        counts.sort_unstable();

        let mut remove:i32 = 0;
        let mut k = k;

        for count in counts.iter() {
            if *count <= k {
                remove += 1;
                k -= count;
            } else {
                break;
            }
        }

        counts.len() as i32 - remove
    }
}